src/Cake/Infrastructure/Scripting/RoslynScriptSession.cs

```diff
@@ -82,6 +99,30 @@ public void ImportNamespace(string @namespace)
 
         public void Execute(Script script)
         {
+            var scriptName = _settings.Script.GetFilename();
+            var cacheDLLFileName = $"{scriptName}.dll";
+            var cacheHashFileName = $"{scriptName}.hash";
+            var cachedAssembly = _scriptCachePath.CombineWithFilePath(cacheDLLFileName);
+            var hashFile = _scriptCachePath.CombineWithFilePath(cacheHashFileName);
+            string scriptHash = default;
+            if (_scriptCacheEnabled && _fileSystem.Exist(cachedAssembly) && !_regenerateCache)
+            {
+                _log.Verbose($"Cache enabled: Checking cache build script ({cacheDLLFileName})");
+                scriptHash = FastHash.GenerateHash(Encoding.UTF8.GetBytes(string.Concat(script.Lines)));
+                var cachedHash = _fileSystem.Exist(hashFile)
+                               ? _fileSystem.GetFile(hashFile).ReadLines(Encoding.UTF8).FirstOrDefault()
+                               : string.Empty;
+                if (scriptHash.Equals(cachedHash, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
`OrdinalIgnoreCase` or just`Ordinal`?

Reply:
FastHash.GenerateHash will always have the same casing and characters, so Ordinal should work.